File: src/main/java/me/testaccount666/serversystem/userdata/vanish/VanishData.java
package me.testaccount666.serversystem.userdata.vanish;

import lombok.Getter;
import lombok.Setter;
import lombok.experimental.Accessors;

@Accessors(fluent = true)
@Getter
@Setter
public class VanishData {
    protected boolean canDrop;
    protected boolean canPickup;
    protected boolean canInteract;
    protected boolean canMessage;

    public VanishData(boolean canDrop, boolean canPickup, boolean canInteract, boolean canMessage) {
        this.canDrop = canDrop;
        this.canPickup = canPickup;
        this.canInteract = canInteract;
        this.canMessage = canMessage;
    }
}

File: src/main/kotlin/me/testaccount666/serversystem/userdata/vanish/VanishData.kt
package me.testaccount666.serversystem.userdata.vanish

import lombok.experimental.Accessors

@Accessors(fluent = true)
class VanishData(
    var canDrop: Boolean,
    var canPickup: Boolean,
    var canInteract: Boolean,
    var canMessage: Boolean
) {
    //TODO: Remove after Kotlin migration!!!!
    fun canDrop(): Boolean = this.canDrop

    fun canPickup(): Boolean = this.canPickup

    fun canInteract(): Boolean = this.canInteract

    fun canMessage(): Boolean = this.canMessage

    fun canDrop(canDrop: Boolean): VanishData {
        this.canDrop = canDrop
        return this
    }

    fun canPickup(canPickup: Boolean): VanishData {
        this.canPickup = canPickup
        return this
    }

    fun canInteract(canInteract: Boolean): VanishData {
        this.canInteract = canInteract
        return this
    }

    fun canMessage(canMessage: Boolean): VanishData {
        this.canMessage = canMessage
        return this
    }
}